packages/core/src/browser/quick-input/quick-input-service.ts

```diff
@@ -0,0 +1,90 @@
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { injectable } from 'inversify';
+import { CancellationToken } from '../../common';
+import URI from '../../common/uri';
+import { QuickInputContribution } from './quick-input-contribution';
+
+export interface QuickPickItem {
+    type?: 'item' | 'separator';
+    id?: string;
+    label: string;
+    meta?: string;
+    ariaLabel?: string;
+    description?: string;
+    detail?: string;
+    resource?: URI;
+    iconClasses?: string[];
+    execute?: (quickPick: any, lookFor: string) => void;
+}
+export interface QuickPickValue<V> extends QuickPickItem {
+    value: V
+}
+
+@injectable()
+export class QuickInputService implements QuickInputContribution {
+    get backButton(): any { return; /* implemented by derived class */ }
+
+    open(filter: string): void { /* implemented by derived class */ }
+
+    registerQuickAccessProvider(): void { /* implemented by derived class */ }
+
+    createInputBox(): any { /* implemented by derived class */ }
+
+    createQuickPick(): any { /* implemented by derived class */ }
+
+    input(options?: any, token?: CancellationToken): Promise<string | undefined> { return Promise.resolve(undefined); /* implemented by derived class */ }
+
+    pick<T extends QuickPickItem>(picks: Promise<any[]> | any[], options?: any, token?: CancellationToken): Promise<T | undefined> {
+        return Promise.resolve(undefined);
+        /* implemented by derived class */
+    }
+
+    showQuickPick<T extends QuickPickItem>(items: Array<T>, options?: any): any { /* implemented by derived class */ }
+
+    hide(): void { /* implemented by derived class */ }
+
+    focus(): void { /* implemented by derived class */ }
+
+    toggle(): void { /* implemented by derived class */ }
+
+    applyStyles(styles: any): void { /* implemented by derived class */ }
+
+    layout(dimension: any, titleBarOffset: number): void { /* implemented by derived class */ }
+
+    navigate?(next: boolean, quickNavigate?: any): void { /* implemented by derived class */ }
+
+    dispose(): void { /* implemented by derived class */ }
+
+    async cancel(): Promise<void> { /* implemented by derived class */ }
+
+    async back(): Promise<void> { /* implemented by derived class */ }
+
+    async accept?(keyMods?: any): Promise<void> { /* implemented by derived class */ }
```

Comment:
> Yes, but you can only rebind what is bound in the first place. Why do we need to bind "QuickInputService" at all? Also, even if we need the dummy implementation, we should not derive the real implementations from it (because of the arguments @marechal-p made)

How would we invoke quickInputService.open('>') for instance without a QuickInputService class and without binding it `frontend-application-module.ts`?

Reply:
You will bind "MonacoQuickInputService" to the service identifier "QuickInputService".

Reply:
Fixed.Added local interfaces to QuickInputService, removed binding of QuickInputService in frontend app